### PR TITLE
fix the bug that outbound.calls.sent is emitted too late

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal

### CODE REVIEW

File: node/out_request.js [hunk 1]
```diff
@@ -301,15 +301,6 @@ TChannelOutRequest.prototype.send = function send(arg1, arg2, arg3, callback) {
         self.span.name = endpoint;
     }
 
-    if (self.retryCount === 0) {
```
Comment (Raynos) [inline node/out_request.js]: this needs to be changed to `if (self.logical === false) {`

All direct requests ( host option ) and streaming requests go through this code path and not through `request.js` 

See how we emit latency stats for an example.

Comment (ShanniLi) [inline node/out_request.js]: fixed.

**Comment by Raynos** [comment]
lgtm.

File: node/request.js [hunk 2]
```diff
@@ -153,6 +153,13 @@ TChannelRequest.prototype.send = function send(arg1, arg2, arg3, callback) {
     self.start = self.timers.now();
     self.resendSanity = self.limit + 1;
 
+    self.channel.outboundCallsSentStat.increment(1, {
+        'target-service': self.serviceName,
+        'service': self.headers.cn,
+        // TODO should always be buffer
```
Comment (kriskowal) [inline node/request.js]: target-endpoint should always be a string. This TODO is not necessary.

**Comment by kriskowal** [comment]
:accept:

File: node/request.js (hunk 2 above)
Comment (ShanniLi) [inline node/request.js]: I remembered @Raynos said that it could be buffer. Can you confirm, Jake?

**Pull request merged**